src/com/dotcms/plugin/saml/v3/rest/api/v1/DotSamlResource.java

```diff
@@ -0,0 +1,346 @@
+                                  @QueryParam(PaginationUtil.FILTER)   final String filter,
+                                  @QueryParam(PaginationUtil.PAGE) final int page,
+                                  @QueryParam(PaginationUtil.PER_PAGE) final int perPage,
+                                  @DefaultValue("upper(name)") @QueryParam(PaginationUtil.ORDER_BY) String orderbyParam,
+                                  @DefaultValue("ASC") @QueryParam(PaginationUtil.DIRECTION) String direction) {
+        final InitDataObject initData = this.webResource.init(null, true, request, true, null);
+        final User user = initData.getUser();
+
+        Response response;
+
+        try {
+            response = this.paginationUtil.getPage(request, user, filter, page, perPage, orderbyParam, direction);
+
+        } catch (Exception e) { // this is an unknown error, so we report as a 500.
+            Logger.error(this,"Error getting idps", e);
+            response = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+        return response;
+    } // getIdps.
+
+    @GET
+    @Path("/idp/{id}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response getIdp(@PathParam("id") final String id, @Context final HttpServletRequest req){
+        final InitDataObject initData = this.webResource.init(null, false, req, false, null);
+        Response response;
+
+        try{
+            final IdpConfig idpConfig = idpConfigHelper.findIdpConfig(id);
+            response = Response.ok(new ResponseEntityView(idpConfig)).build();
+        } catch (DotDataException e) {
+            Logger.error(this,"Idp not found (" + e.getMessage() + ")", e);
+            response = ExceptionMapperUtil.createResponse(null, "Idp not found (" + e.getMessage() + ")");
+        } catch (IOException e) {
+            Logger.error(this,"Idp is not valid (" + e.getMessage() + ")", e);
+            response = ExceptionMapperUtil.createResponse(null, "Idp is not valid (" + e.getMessage() + ")");
+        } catch (JSONException e) {
+            Logger.error(this,"Error handling json (" + e.getMessage() + ")", e);
+            response = ExceptionMapperUtil.createResponse(null, "Error handling json (" + e.getMessage() + ")");
+        } catch (Exception e) { // this is an unknown error, so we report as a 500.
+            Logger.error(this,"Error getting posting idp", e);
+            response = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+
+        return response;
+    } //getIdp.
+
+    @POST
+    @Path("/idp")
+    @JSONP
+    @NoCache
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response createIdpConfig(@Context final HttpServletRequest req,
+                                          @FormDataParam("id") String id,
```

Comment:
it would be easy just to encapsulate it into a Form

Reply:
do you have an example of this?